### ISSUE TITLE
scene_tree_editor selectable item color set to disabled before filter updates node tree

### ISSUE THREAD

**Issue by jacwilso** [godotengine/godot#81824]
### Godot version

4.1

### System information

macOS

### Issue description

Selectable nodes in the scene_tree_editor have color set to disabled when the window is first opened. Only after modifying the filter will the nodes be correctly colored as selectable.

Current:
<img width="1748" alt="Screenshot 2023-09-17 at 12 22 32 PM" src="https://github.com/godotengine/godot/assets/21345963/5ee5e01a-0000-4e40-b991-794519dead5a">

After filter:
<img width="1748" alt="Screenshot 2023-09-17 at 12 22 37 PM" src="https://github.com/godotengine/godot/assets/21345963/f08ec6ff-5156-4c33-9e8f-e6af5942c052">

Expected (before applying any filter):
<img width="1748" alt="Screenshot 2023-09-17 at 12 13 24 PM" src="https://github.com/godotengine/godot/assets/21345963/3bbb3d7a-6dbc-4039-9089-60e091aeb9ea">

### Steps to reproduce

1. Create GDExtension class with a variable of `NodePath` type
2. Bind that variable to be exposed in the editor -- use `PROPERTY_HINT_NODE_PATH_VALID_TYPES` to hint the valid types, ie `ClassDB::add_property("TestNode", PropertyInfo(Variant::NODE_PATH, "gd_example", PROPERTY_HINT_NODE_PATH_VALID_TYPES, "GDExample"), "set_gd_example", "get_gd_example");`
3. In the editor create a node with the GDExtension class and 2 other nodes, 1 with and 1 without the hinted type
4. In the editor open the assign, you will see all nodes are colored as disabled, even the selectable/hinted node

### Minimal reproduction project

N/a -- reproduction is trivial, proposed solution below

```
diff --cc editor/gui/scene_tree_editor.cpp
index fbe167814d,1f9eefb3cf..0000000000
--- a/editor/gui/scene_tree_editor.cpp
+++ b/editor/gui/scene_tree_editor.cpp
@@@ -263,9 -267,26 +264,30 @@@ void SceneTreeEditor::_add_nodes(Node *
  		}
  		item->set_text(0, node_name);
  		item->set_selectable(0, marked_selectable);
		_set_item_custom_color(item, get_theme_color(SNAME("accent_color"), EditorStringName(Editor)));
  	} else if (!p_node->can_process()) {
+ 		bool is_selectable = false;
+ 		if (!valid_types.is_empty()) {
+ 			for (const StringName &E : valid_types) {
+ 				if (p_node->is_class(E) ||
+ 						EditorNode::get_singleton()->is_object_of_custom_type(p_node, E)) {
+ 					is_selectable = true;
+ 					break;
+ 				}
+ 			}
+ 		}
+ 
+ 		if (!is_selectable) {
+		        _set_item_custom_color(item, get_theme_color(SNAME("disabled_font_color"), EditorStringName(Editor)));
+ 		}
  	} else if (!marked_selectable && !marked_children_selectable) {
  		Node *node = p_node;
  		while (node) {
```

### Regression
Caused by this commit: https://github.com/godotengine/godot/commit/083aa9b95e6e32d13bf6a406137b3fd12287c4d0

**Comment by akien-mga** [comment]
CC @godotengine/docks

**Comment by KoBeWi** [comment]
Fixed in 4.2
